### ISSUE TITLE
Can't upgrade to netlify-cli 17.19.3 (or later), fails with "Cannot find package 'regexp-tree' ..."

### ISSUE THREAD

**Issue by pogo19** [netlify/cli#6444]
### Describe the bug

Upgrade of globally installed CLI (version 17.19.0) to 17.19.3:
```
pogo@narsil:~$ yarn global upgrade-interactive --latest
yarn global v1.22.19
info Color legend : 
 "<red>"    : Major Update backward-incompatible updates 
 "<yellow>" : Minor Update backward-compatible features 
 "<green>"  : Patch Update backward-compatible bug fixes
? Choose which packages to update. 
? Choose which packages to update. netlify-cli@17.19.3
info Installing "dependencies"...
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning "netlify-cli > @netlify/build > ts-node@10.9.1" has unmet peer dependency "@types/node@*".
[4/4] Building fresh packages...
[-/12] ⠈ waiting...




Error: /home/pogo/.config/yarn/global/node_modules/netlify-cli: Command failed.
Exit code: 1
Command: node ./scripts/postinstall.js
Arguments: 
Directory: /home/pogo/.config/yarn/global/node_modules/netlify-cli
Output:
Success! Netlify CLI has been installed!

Your device is now configured to use Netlify CLI to deploy and manage your Netlify sites.

Next steps:

  netlify init     Connect or create a Netlify site from current directory
  netlify deploy   Deploy the latest changes to your Netlify site

For more information on the CLI run netlify help
Or visit the docs at https://cli.netlify.com


 ›   Error: Netlify CLI has terminated unexpectedly
This is a problem with the Netlify CLI, not with your application.
If you recently updated the CLI, consider reverting to an older version by running:

npm install -g netlify-cli@VERSION

You can use any version from https://ntl.fyi/cli-versions.

Please report this problem at https://ntl.fyi/cli-error including the error details below.

Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'regexp-tree' imported from /home/pogo/.config/yarn/global/node_modules/@netlify/build/node_modules/@netlify/edge-bundler/dist/node/declaration.js
    at packageResolve (node:internal/modules/esm/resolve:853:9)
    at moduleResolve (node:internal/modules/esm/resolve:910:20)
    at defaultResolve (node:internal/modules/esm/resolve:1130:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:396:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:365:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:240:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:85:39)
    at link (node:internal/modules/esm/module_job:84:36)

  System:
    OS: Linux 6.5 Ubuntu 23.10 23.10 (Mantic Minotaur)
    CPU: (16) x64 13th Gen Intel(R) Core(TM) i7-1360P
  Binaries:
    Node: 20.11.1 - /tmp/yarn--1710504580642-0.4224374161530051/node
    Yarn: 1.22.19 - /tmp/yarn--1710504580642-0.4224374161530051/yarn
    npm: 10.2.4 - /usr/bin/npm
  Browsers:
    Chrome: 122.0.6261.128
    Firefox: 123.0.1
    at ProcessTermError.ExtendableBuiltin (/usr/share/yarn/lib/cli.js:721:66)
    at ProcessTermError.MessageError (/usr/share/yarn/lib/cli.js:750:123)
    at new ProcessTermError (/usr/share/yarn/lib/cli.js:790:113)
    at ChildProcess.<anonymous> (/usr/share/yarn/lib/cli.js:25787:17)
    at ChildProcess.emit (node:events:518:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5)
```

### Steps to reproduce

Run `yarn global upgrade-interactive --latest` to upgrade from globally installed 17.19.0 to 17.19.3.

### Environment

```
  System:
    OS: Linux 6.5 Ubuntu 23.10 23.10 (Mantic Minotaur)
    CPU: (16) x64 13th Gen Intel(R) Core(TM) i7-1360P
    Memory: 20.14 GB / 30.92 GB
    Container: Yes
    Shell: 5.2.15 - /bin/bash
  Binaries:
    Node: 20.11.1 - /usr/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 10.2.4 - /usr/bin/npm
```

**Comment by pogo19** [comment]
The problem is still there even in 17.19.4

**Comment by pogo19** [comment]
17.19.6 too

**Comment by pogo19** [comment]
Still there in 17.20.0

**Comment by pogo19** [comment]
17.20.1 as well

**Comment by pogo19** [comment]
17.21.1 still affected

**Comment by pogo19** [comment]
Fixed in version 17.21.2.
